servers/cu/src/domain/client/ao-process.js

```diff
@@ -414,6 +417,97 @@ export function writeProcessMemoryFileWith ({ DIR, writeFile }) {
   }
 }
 
+/**
+ * ################################
+ * #########  sqlite utils ##########
+ * ################################
+ */
+
+export function findCheckpointRecordBeforeWith ({ db }) {
+  function createQuery ({ processId, before }) {
+    const timestamp = before === 'LATEST' ? new Date().getTime() : before.timestamp
+    return {
+      sql: `
+        SELECT *
+        FROM ${CHECKPOINTS_TABLE}
+        WHERE
+          processId = ? AND timestamp < ?
+        ORDER BY timestamp DESC
+        LIMIT 5;
```

Comment:
I'd place a comment explaining the rationale behind this query. Why `LIMIT 5`? Why both `processId` AND `timestamp`? How do this relate to the `latestCheckpointBefore` subsequent step.